### DISCUSSION TITLE
Getting started with Horizon and facing issues using the Swell storefront editor.

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#271] [Help] [answered]
Hi, we've been facing issues trying to use the Horizon them and the new updates to the Swell storefront editor. Basically, the editor doesn't seem to maintain the changes we make when using it. If we add a menu to the header using the editor it works for a time, until we add a footer, then the header menu will mysteriously disappear. Or sometimes the changes don't work at all, there's not a clear pattern I can find, there's also no clear errors in the console.  Basically any changes we made using the editor seem transitory and something doesn't seem to work right.

We followed all the documentation, deployed to vercel, set `NEXT_PUBLIC_SWELL_EDITOR=true` etc.

It appears to work within the swell editor, changes get reflected for some time, then they get reverted as we make other changes.

Is there something here we're missing?

Any pointers would greatly appreciated.

I also made a video in the issue here: https://github.com/swellstores/horizon/issues/48

**Comment by sags95** [comment]
Thank you for bringing this to our attention @holden, I've passed this on to the team so we can investigate. It's possible that there is some issue here rather than an misconfiguration.

**Reply by holden** [reply]
Awesome, thanks. I set it up here on vercel: [beta.cirqus.co](https://beta.cirqus.co/categories/workshops) with the store_id: `cirqus-marketplace` if that helps.

I noticed a couple of other things as well:

* it's hitting quite a few 403 json errors, (see the image below) and as a result it loads the data quite slowly between requests, i'm pretty sure this isn't anything I did as it appeared to do the same even when I created a new instance from scratch.
* also when you change the currency from the default of USD (eg to EUR or GBP), it permanently removes USD from being an option so you can't go back to it again later

<img width="565" alt="CleanShot 2023-06-14 at 11 23 06@2x" src="https://github.com/swellstores/community/assets/48241/7f4c2eb7-a835-449a-9415-98c3bd51589e">

Thanks again

**Reply by logeshswell** [reply]
* 403 error is being thrown since the fetchCart returns null [here](https://github.com/swellstores/horizon/blob/main/pages/api/cart/index.ts#L60)
* The following error 'Invalid payload' also correspond to the same line as `JSON.parse(string)` throws an error and is being logged [here](https://github.com/swellstores/horizon/blob/main/stores/cart.ts#L60)
  
So, these can be modified as per your need.

**Reply by logeshswell** [reply]
@holden Currency dropdown issue is fixed in your storefront now. Please check and confirm.

<img width="1512" alt="Screenshot 2023-06-28 at 3 55 40 PM" src="https://github.com/swellstores/community/assets/100766131/8304dafe-fa09-4422-ae3b-9dc9ddb06a75">

**Reply by holden** [reply]
Awesome, thanks. Yes it works fine now. How was this fixed? Was it just a simple setting I missed somewhere?

Regarding the 403s, so you're saying this is expected behavior?  As I was only noticing it because I was having the error with the navigation.

**Reply by logeshswell** [reply]
> Was it just a simple setting I missed somewhere?

No, the currency data were corrupted in the backend of your storefront. I just have to refresh it.

> so you're saying this is expected behavior?

As per the codebase, yes it is expected and you can update it to return 404 or 200 status code as per your wish.

**Comment by holden** [comment]
Hi @sags95 can I get an update on this? It still appears to be not working properly. Thanks

**Reply by Strongeyed** [reply]
@holden Maybe this temporary fix for the issue described (pushed recently to the Swell repo) by applying the following in the sync script `-s defaults`:

<img width="1200" alt="248935651-ff102f16-a9e2-4f79-8939-6940b8e87179" src="https://github.com/swellstores/community/assets/128098520/2e22dee2-c815-47c0-bd4e-88ec3664dab5">

**Reply by holden** [reply]
No, I checked this was already line appeared to be already added in the version of horizon I'm using. Thanks though, I'm not sure what would cause adding a footer to remove a header and vice versa. ;-(

**Reply by logeshswell** [reply, marked as answer]
@holden I noticed 2 issues in your storefront navigation.
1. There shouldn't be more than 1 record of the same **Theme area** (Header/Footer menu)
2. Menu **Alias** should be `header` and `footer` for **Theme area** (Header and Footer menu respectively).

<img width="917" alt="Screenshot 2023-06-28 at 4 06 55 PM" src="https://github.com/swellstores/community/assets/100766131/2a70400e-d553-4ccb-b0c3-25c90e83ef12">

<img width="917" alt="Screenshot 2023-06-28 at 4 07 03 PM" src="https://github.com/swellstores/community/assets/100766131/2aee90e7-e41e-479d-986e-998b8dda8ebf">

Secondly, there is a bug found in the storefront editor while updating the header and footer. As an immediate workaround, I fixed it by running `npm install` inside the horizon theme for your storefront and you could see both the header and footer being displayed in your storefront now. Please check and confirm.

<img width="1385" alt="Screenshot 2023-06-28 at 4 11 59 PM" src="https://github.com/swellstores/community/assets/100766131/fd97d716-c19d-41f1-a854-b4c4de2ee6dd">

<img width="1386" alt="Screenshot 2023-06-28 at 4 12 05 PM" src="https://github.com/swellstores/community/assets/100766131/48e25a46-b603-43a6-b3d4-fdcf42bd13a0">

**Reply by holden** [reply]
Nope, this still doesn't work for me. I made some very minor changes and immediately the footer and header disappeared.

The repo I deployed to vercel is here, you can see there haven't been any real changes.

https://github.com/razrfly/cirqus-horizon/

**Reply by logeshswell** [reply]
@holden As I mentioned above, there is a UI bug found in the storefront editor while updating the header or footer.
It updates header or footer at once but not both. I will file a bug ticket for faster resolution.

When you encounter the same issue, running `npm install` will fix this issue as temporary workaround.

**Reply by holden** [reply]
Please do, thanks for that. Running `npm install` does return both the menus, it also seems put everything else back to default too, like the logo, etc. But I can work with that for now.  Thanks

**Reply by logeshswell** [reply]
To avoid restoring with the default settings, please pass `-s defaults` flag in the sync script for **editor** and **defaults** commands as @Strongeyed mentioned in [this comment](https://github.com/orgs/swellstores/discussions/271#discussioncomment-6290293).

**Reply by holden** [reply]
Ok, now it's nearly perfect. Thanks again.

**Reply by logeshswell** [reply]
This UI bug is fixed now. Can you check and confirm?

**Comment by holden** [comment]
One other problem I've noticed is on the category details page, whenever I try to make small changes there. It shows them auto-update, however it doesn't persist them

I was trying to change it to a 5 instead of 4 products per row and add several featured categories to be above the products, but none of the changes updated anything.

https://github.com/swellstores/community/assets/48241/c0464a41-b15f-4d3d-bcf9-7f1cf9ed05d8

**Reply by logeshswell** [reply]
It shouldn't be a problem anymore, as I can see that you switched your frontend from static-mode to server-side rendering.
It helps to render the content immediately when the page is reloaded without the need to build the project.
